certbot-ci/certbot_integration_tests/rfc2136_tests/context.py

```diff
@@ -0,0 +1,70 @@
+from contextlib import contextmanager
+from pytest import skip
+from pkg_resources import resource_filename
+import tempfile
+
+from certbot_integration_tests.certbot_tests import context as certbot_context
+from certbot_integration_tests.utils import certbot_call
+
+
+class IntegrationTestsContext(certbot_context.IntegrationTestsContext):
+    """Integration test context for certbot-dns-rfc2136"""
+    def __init__(self, request):
+        super(IntegrationTestsContext, self).__init__(request)
+
+        self.request = request
+
+        self._dns_xdist = None
+
+        if hasattr(request.config, 'slaveinput'):  # Worker node
+            self._worker_id = request.config.slaveinput['slaveid']
+            self._dns_xdist = request.config.slaveinput['dns_xdist']
+        elif 'dns_xdist' in request.config:  # Primary node
+            self._worker_id = 'primary'
```

Comment:
This is already done in the parent class for a different condition (`not hasattr(request.config, 'slaveinput')`).